internal/regex_helpers.go

```diff
@@ -0,0 +1,39 @@
+package internal
+
+import "regexp"
+
+// MatchNamedCaptureGroups takes a regular expression and string and returns all of the named capture group results in a map.
+// Note: this is only for the first match in the regex.
+func MatchNamedCaptureGroups(regEx *regexp.Regexp, content string) map[string]string {
+	// note: we are looking across all matches and stopping on the first non-empty match. Why? Take the following example:
+	// input: "cool something to match against" pattern: `((?P<name>match) (?P<version>against))?`. Since the pattern is
+	// encapsulated in an optional capture group, there will be results for each character, but the results will match
+	// on nothing. The only "true" match will be at the end ("match against").
+	allMatches := regEx.FindAllStringSubmatch(content, -1)
+	for matchIdx, match := range allMatches {
+		// fill a candidate results map with named capture group results, accepting empty values, but not groups with
+		// no names
+		results := make(map[string]string)
+		for nameIdx, name := range regEx.SubexpNames() {
+			if nameIdx <= len(match) && len(name) > 0 {
+				results[name] = match[nameIdx]
```

Comment:
yup yup, since the intention is that the return is a mapping, and the mapping key is the capture group name, there can be only one value. My take: callers shouldn't be providing regexes with multiple capture groups with the same name.

Reply:
Okay cool, that makes sense. In that case, I'd leave just a "nit" here:

>callers shouldn't be providing regexes with multiple capture groups with the same name

I'd say this explicitly in the doc comment, since the caller would have no other way to find that out until runtime.

Reply:
done in 527dc39c82857f6df5717bb213febfa4f1e1c36d